### PR TITLE
fix(terms): Add description on pages, update links

### PR DESCRIPTION
**Pull request by cmp5987**
/kind cleanup
/kind documentation

#### What this PR does / why we need it:

- Adds description to hosts/tasks/quests
- Changes nav bar items
- Changes links to tome now that the documentation exists
- Updates doc language on term page to match descriptions

### CODE REVIEW

File: docs/_docs/user-guide/terminology.md [hunk 1]
```diff
@@ -8,36 +8,36 @@ permalink: user-guide/terminology
...
+Throughout the documentation terms that are used to reference various components of Realm. Below we attempt to define some of those terms, to add some clarity to that other documentation.
 
 ### Implant
 
 References malicious code or persistence mechanisms that are deployed to compromise target systems. Imix is the primary implant used with Realm.
 
 ### Agent
 
-An Agent is a type of implant which retrieves execution instructions by connecting to our backend infrastructure (calling back) and querying for new tasks.
+An agent is a compiled program.
```
Comment (hulto) [inline docs/_docs/user-guide/terminology.md:19]: I think we need to expand on this not sure if we wanna do that in this PR

File: docs/_docs/user-guide/terminology.md [hunk 2]
```diff
@@ -8,36 +8,36 @@ permalink: user-guide/terminology
...
 
 ### Beacon
 
-A Beacon is a running instance of an Agent. A Host may have multiple active Beacons that use the same underlying Agent.
+A Beacon is an instance of an Agent running in a process. Beacons are the underlying unit that can be tasked with instructions to execute.
```
Comment (hulto) [inline docs/_docs/user-guide/terminology.md:23]: Running as a process

File: docs/_docs/user-guide/terminology.md [hunk 3]
```diff
@@ -8,36 +8,36 @@ permalink: user-guide/terminology
...
+Hosts are in-scope systems for the current engagement. A host can have multiple beacons which can execute instructions provided by tomes.
 
 ### Quest
 
-A Quest represents a collection of tasks, each with a unique host. This is how Realm enables multi-host management.
+Quests enable multi-beacon managment by taking a list of beacons and executing a tome with customized parameters against them. A quest is made up of tasks assocaited with a single beacon.
+
+### Task
+
+A task is a single instance of a tome plus its parameters executed against a single beacon.
```
Comment (hulto) [inline docs/_docs/user-guide/terminology.md:35]: I'd add back the "For example...listing files...one Task
per beacon...execution output.

File: docs/_docs/user-guide/terminology.md [hunk 4]
```diff
@@ -8,36 +8,36 @@ permalink: user-guide/terminology
...
 
 ### Eldritch
 
 Eldritch is our Pythonic Domain Specific Language (DSL), which can be used to progammatically define red team operations. Many of the language's built-in features do not rely on system binaries. For more information, please see the [Eldritch section](/user-guide/eldritch) of the documentation.
 
 ### Tome
 
-A Tome is a prebuilt Eldritch bundle, which provides execution instructions to a Beacon. Tomes can embed files and accept parameters to change their behavior at runtime. Tavern's built-in Tomes are defined [here](https://github.com/spellshift/realm/tree/main/tavern/tomes).
+A Tome is a prebuilt Eldritch bundle, which includes execution instructions and files. Tomes are how beacon actions are defined and change their behavior at runtime. Tavern's built-in Tomes are defined [here](https://github.com/spellshift/realm/tree/main/tavern/tomes).
```
Comment (hulto) [inline docs/_docs/user-guide/terminology.md:43]: I'd add back the embedded files.

**Review by hulto — COMMENTED** [review]
Docs thoughts

File: docs/_docs/user-guide/terminology.md (hunk 1 above)
Comment (cmp5987) [reply docs/_docs/user-guide/terminology.md:19]: Should I leave it as it was or keep the change that we discussed?

Comment (cmp5987) [reply docs/_docs/user-guide/terminology.md:19]: This simple sentence what what we came up with about 2hrs ago. I was asking if we should revert back to the expansive text.

**Review by hulto — DISMISSED** [review]
Please swap agent term back to OG

**Review by hulto — APPROVED** [review]

**Pull request merged**